grafana/data_source_folder_test.go

```diff
@@ -0,0 +1,33 @@
+//go:build oss
+// +build oss
+
+package grafana
+
+import (
+	"testing"
+
+	gapi "github.com/grafana/grafana-api-golang-client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccDatasourceFolder(t *testing.T) {
+	var folder gapi.Folder
+	checks := []resource.TestCheckFunc{
+		testAccFolderCheckExists("grafana_folder.test", &folder),
+		resource.TestCheckResourceAttr(
+			"data.grafana_folder.from_title", "title", "test-folder",
+		),
```

Comment:
Can we add tests to check uid and id?

```suggestion
		resource.TestCheckResourceAttr(
			"data.grafana_folder.from_title", "title", "test-folder",
		),
		resource.TestMatchResourceAttr(
			"data.grafana_folder.from_title", "id", regexp.MustCompile(`^\d+$`),
		),
		resource.TestMatchResourceAttr(
			"data.grafana_folder.from_title", "uid", regexp.MustCompile(`^[a-zA-Z0-9-]+$`),
		),
```